packages/mui-base/src/Tooltip/Arrow/TooltipArrow.tsx

```diff
@@ -0,0 +1,86 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import type { TooltipArrowOwnerState, TooltipArrowProps } from './TooltipArrow.types';
+import { useTooltipPopupContext } from '../Popup/TooltipPopupContext';
+import { tooltipArrowStyleHookMapping } from './styleHooks';
+import { useComponentRenderer } from '../../utils/useComponentRenderer';
+import { useForkRef } from '../../utils/useForkRef';
+import type { GenericHTMLProps } from '../../utils/BaseUI.types';
+
+/**
+ * The tooltip arrow caret element.
+ *
+ * Demos:
+ *
+ * - [Tooltip](https://mui.com/base-ui/react-tooltip/)
+ *
+ * API:
+ *
+ * - [TooltipArrow API](https://mui.com/base-ui/react-tooltip/components-api/#tooltip-arrow)
+ */
+const TooltipArrow = React.forwardRef(function TooltipArrow(
+  props: TooltipArrowProps,
+  forwardedRef: React.ForwardedRef<HTMLDivElement>,
+) {
+  const { className, render, hideWhenUncentered = false, ...otherProps } = props;
+
+  const { open, arrowRef, side, alignment, arrowUncentered, getArrowProps } =
+    useTooltipPopupContext();
+
+  const ownerState: TooltipArrowOwnerState = React.useMemo(
+    () => ({
+      open,
+      side,
+      alignment,
+    }),
+    [open, side, alignment],
+  );
+
+  const mergedRef = useForkRef(arrowRef, forwardedRef);
+
+  const { renderElement } = useComponentRenderer({
+    propGetter: (externalProps: GenericHTMLProps = {}) =>
+      getArrowProps({
+        ...externalProps,
+        style: {
```

Comment:
You should be able to write `propGetter: getArrowProps` and pass in the style prop to the `extraProps` field. Let me know if this doesn't work.